bytestream.go

```diff
@@ -73,22 +77,46 @@ func ByteStreamConsumer(opts ...byteStreamOpt) Consumer {
 		}
 		b := buf.Bytes()
 
-		if bu, ok := data.(encoding.BinaryUnmarshaler); ok {
-			return bu.UnmarshalBinary(b)
-		}
+		switch dest := data.(type) {
+		case encoding.BinaryUnmarshaler:
+			return dest.UnmarshalBinary(b)
+		case *string:
+			*dest = string(b)
+
+			return nil
+		case *[]byte:
+			*dest = b
+
+			return nil
+		case *any:
+			switch (*dest).(type) {
+			case string:
+				*dest = string(b)
+
+				return nil
+
+			case []byte:
+				*dest = b
 
-		if data != nil {
-			if str, ok := data.(*string); ok {
-				*str = string(b)
 				return nil
 			}
-		}
+		default:
```

Comment:
this default clause probably covers the typeswitch above

Reply:
yes I know. This was an attempt to catch common cases without reflection, but type switch instead.
The belief is that type switch acts faster, but I am not really convinced. Hence the experiment.

Reply:
Unfortunately, we can't yet assert the underlying type with something like "case ~string".

Reply:
My idea was to skip the allocation of a reflect.Value for the most common paths, leaving the call to "reflect.ValueOf(data)" to the only cases when the type is realiased by the caller.
I could investigate deeper about the real benefit of such an approach.